[Release Guide] Document Release Candidates (#725)
submariner-io/submariner-website: src/content/development/contribution-guide/_index.en.md

#### Release Candidates

At the end of 12 weeks, the project is ready to be released and the pre-release `rc0` is created.
At this point, as detailed in the [release process] documentation, stable branches are created and the project goes into feature freeze
for the release branches.

> Two [test days](#test-days) will take place after the release is created, as the project members make sure the release is ready to be
> [generally available](#general-availability).

Reviewer: ```suggestion
Two [test days](#test-days) will take place after the release is created, as the project members make sure the release is ready for
[general availability](#general-availability).
```

Any bugs found during the test days will need to be labeled with the appropriate `testday` label.
The project members will triage the test day bugs and identify any high priority ones that should be addressed before general availability.

If any high priority bugs were identified after `rc0`, a new release `rc1` will be planned to allow for fixing them.

> The `rc1` release will be planned according to the team's discretion and has no expected date.

Reviewer: ```suggestion
The `rc1` release will be planned at the team's discretion and has no expected date.
```
